src/bcf/header.rs

```diff
@@ -14,6 +14,26 @@ use htslib;
 pub type SampleSubset = Vec<i32>;
 
 
+custom_derive! {
+    /// A newtype for IDs from BCF headers.
+    #[derive(
+        NewtypeFrom,
+        NewtypeDeref,
+        NewtypeAdd(*),
+        NewtypeSub(*),
+        NewtypeMul(*),
+        NewtypeDiv(*),
+        PartialEq,
+        PartialOrd,
+        Copy,
+        Clone,
+        Debug,
+        Default
+    )]
+    pub struct Id(pub u32);
+}
```

Comment:
I think I would probably not derive Add, Sub, Mul, Div and Default, as there is no need for these, and they could only be used to generate invalid ids. However, I can also change this after merging